lib/hls/hls_classes.js

```diff
@@ -75,15 +75,6 @@ shaka.hls.PlaylistType = {
  * @struct
  */
 shaka.hls.Tag = function(id, name, attributes, value = null) {
-  goog.asserts.assert(
-      (attributes.length == 0 && value) ||
-      (attributes.length > 0 && !value) ||
-      (attributes.length == 0 && !value),
-      'Tags can only take the form ' +
-      '(1) <NAME>:<VALUE> ' +
-      '(2) <NAME>:<ATTRIBUTE_LIST> ' +
-      ' (3) <NAME>');
-
   /** @const {number} */
   this.id = id;
 
```

Comment:
A bit below this, Tag has a "toString" method. That method was also made assuming that tags cannot have both values and attribute lists. It should also be updated to account for this change.

Correspondingly, there's also a few simple tests for "toString" in the manifest text parser unit tests. There should be a new test for attribute+value tags in that.

Reply:
Good catch. Been a while since I've used protoypes directly. The new test case also caught that `hls.Tag.toString()` is treating all attributes as strings and stringifying them with quotes, so I added numeric value detection to avoid adding quotes. Detecting a numeric string with `isNan(Number(str))` is shaky but all the ways of tricking it that I know shouldn't happen in this case.